### PR TITLE
#3: Step 2 (closes #3)

### LINKED ISSUE #3: Step 2
## requirements
- In case you used an `if` or a nested `match` in Step 1, refactor your code to use a single match on a tuple `(userMove, computerMove)`

## specifications

- Minor refactor, using pattern matching instead of an if/else

### Relevant points from the solution

- we’re creating a tuple to match on → this makes the pattern match very convenient
- we’re using a “guard” to ensure `x` and `y` are equal in the Draw case
  - if you’re thinking whether `case (x, x)` would’ve worked, nope, it doesn’t, sorry. (It works in Prolog though!)
- `case _` as the default case → this makes the pattern match complete (i.e. it covers all the cases), which is always a good idea
- using “0”, “1”, “2” still sucks

### PR DESCRIPTION
**Pull request by raasoft**
Closes #3

### tests performed

kapture 2018-07-11 at 12 22 04

### CODE REVIEW

File: src/main/scala/rps/Game.scala [hunk 1]
```diff
@@ -25,28 +25,30 @@ object Game {
...
+        e.g.
+        1. case ("0") => action a
+        2. case ("0") => action b
+
+        1 and 2 matches, but action a is executed.
+        */
+    (userMove, cpuMove) match {
+      case (x,y) if (x == y) => println("It's a draw! 🧐") // I used a "guard"
+      case ("1","0") | ("2","1") | ("0","2") => println("You Win! 😤") // I used pipe to match multiple conditions
+      case _ => println("You Lose! 🤩") // _ stands for the default case
```
Comment (gabro) [inline src/main/scala/rps/Game.scala:40]: very good use of pattern matching, with guards and pipes 💯

File: src/main/scala/rps/Game.scala [hunk 2]
```diff
@@ -25,28 +25,30 @@ object Game {
...
-      "📄"
-    } else if (move == "2") {
-      "✂️"
-    } else {
-      ""
+    move match {
+      case "0" => "💎"
+      case "1" => "📄"
+      case "2" => "✂️"
+      case _ => "💩"
```
Comment (gabro) [inline src/main/scala/rps/Game.scala:51]: 💩 is a very funny default, but as I mentioned, we'll do better ;)

**Review by gabro — APPROVED** [review]
👍 nice!

File: src/main/scala/rps/Game.scala (hunk 2 above)
Comment (raasoft) [reply src/main/scala/rps/Game.scala:51]: ahah! Unfortunately I knew his time was somehow short, that's why I decided to give him a small role in our program before saying goodbye!

**Pull request merged**